### DISCUSSION TITLE
GitHub Actions Accessibility

### DISCUSSION THREAD

**Discussion by oddsorevans** [community/community#70559] [Actions] [answered]
Is there a way to make a GitHub Action accessible for anyone to run it? In my use case, I have some python code that gathers data and puts it into a spreadsheet, and I wanted to make that accessible through a GHA without the need to download or fork the repo. All of the documentation I've read about Action Permissions have to do with Fork of the repo or collaborators.

**Comment by spenserblack** [comment, marked as answer]
You might want to check the events that can trigger workflows: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows. Anyone who has permission to cause one of these events effectively has permission to run the workflow. However, GitHub's terms of service[^1] state
>Actions should not be used for ... any activity that places a burden on our servers, where that burden is disproportionate to the benefits provided to users (for example, don't use Actions as a content delivery network or as part of a serverless application, but a low benefit Action could be ok if it’s also low burden)

I'm not really sure if using actions in this way violates the ToS or not, but I'd recommend looking into other options. For example, a web service where users can generate and download the spreadsheet.

[^1]: https://docs.github.com/en/site-policy/github-terms/github-terms-for-additional-products-and-features#actions

**Reply by oddsorevans** [reply]
Thank you this is what I was looking for. I considered doing a web service if there wasn't a way, and I think the ToS just answered that for me.
